src/org/labkey/test/components/ui/navigation/FindByIdsDialog.java

```diff
@@ -0,0 +1,82 @@
+package org.labkey.test.components.ui.navigation;
+
+import org.apache.commons.lang3.StringUtils;
+import org.labkey.test.Locator;
+import org.labkey.test.components.bootstrap.ModalDialog;
+import org.labkey.test.components.html.RadioButton;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+
+public class FindByIdsDialog  extends ModalDialog
+{
+    public static final String TITLE = "Find Samples";
+
+    public FindByIdsDialog(WebDriver driver)
+    {
+        super(new ModalDialog.ModalDialogFinder(driver).withTitle(TITLE));
+    }
+
+    public boolean isBarcodeChecked()
+    {
+        return elementCache().barcodeRadio.isChecked();
+    }
+
+    public boolean isSampleIdChecked()
+    {
+        return elementCache().sampleIDsRadio.isChecked();
+    }
+
+    public FindByIdsDialog chooseBarcodes()
+    {
+        elementCache().barcodeRadio.check();
+        return this;
+    }
+
+    public FindByIdsDialog chooseSampleIDs()
+    {
+        elementCache().sampleIDsRadio.check();
+        return this;
+    }
+
+    public FindByIdsDialog addIds(List<String> ids)
+    {
+        elementCache().idTextArea.sendKeys(StringUtils.join(ids, "\n"));
+        return this;
+    }
+
+    public void clickCancel()
+    {
+        elementCache().cancelButton.click();
+    }
+
+    public void clickFindSamples()
+    {
+        elementCache().findSamplesButton.click();
+    }
+
+    @Override
+    protected FindByIdsDialog.ElementCache newElementCache()
+    {
+        return new FindByIdsDialog.ElementCache();
+    }
+
+    @Override
+    protected FindByIdsDialog.ElementCache elementCache()
+    {
+        return (FindByIdsDialog.ElementCache) super.elementCache();
+    }
+
+    protected class ElementCache extends ModalDialog.ElementCache
+    {
+        RadioButton barcodeRadio = RadioButton.RadioButton(Locator.radioButtonByName("uniqueIds")).findWhenNeeded(getComponentElement());
+        RadioButton sampleIDsRadio = RadioButton.RadioButton(Locator.radioButtonByName("sampleIds")).findWhenNeeded(getComponentElement());
+        WebElement idTextArea = Locator.tag("textarea").findWhenNeeded(getComponentElement());
+
+        WebElement cancelButton = Locator.tagWithText("button", "Cancel")
+                .findWhenNeeded(getComponentElement());
+        WebElement findSamplesButton = Locator.tagWithText("button", "Find Samples")
+                .findWhenNeeded(getComponentElement());
```

Comment:
General best practice is to mark elements in the ElementCache as final (I don't always follow best practices).